tests/property/specification_test.py

```diff
@@ -57,9 +54,7 @@ def test_parsing_boolean_expressions(expression: expr.Expr) -> None:
         ),
         strategies.boolean_expressions(
             st.one_of(
-                strategies.aggregates(strategies.numbers())
-                | strategies.strings()
-                | strategies.mathematical_expressions(
+                strategies.mathematical_expressions(
```

Comment:
What is the reason for this change?

Reply:
The `extended_expression` rule of the new parser which is used here, does not parse arrays or strings anymore. Hence, the generators had to be removed.

Reply:
This would mean the new parser would be unable to parse expressions where one part of a relation is an aggregate or a string. From what I can see that is not the case. Could you please recheck that?

Reply:
As discussed separately, the actual reason is that the new parser will reject binary operations other than `=` and `/=` for aggregates/strings. I factored out those tests in 68ece9e0f0e43cd40790fcc290dbbebfe7d4c108.